Sources/PostgresNIO/New/Data/Array+PostgresCodable.swift

```diff
@@ -85,6 +85,18 @@ extension UUID: PostgresArrayEncodable {
     public static var psqlArrayType: PostgresDataType { .uuidArray }
 }
 
+extension Range: PostgresArrayDecodable where Bound: PostgresRangeBound {}
+
+extension Range: PostgresArrayEncodable where Bound: PostgresRangeBound {
+    public static var psqlArrayType: PostgresDataType { Bound.rangeType.arrayType! }
+}
+
+extension ClosedRange: PostgresArrayDecodable where Bound: PostgresRangeBound {}
+
+extension ClosedRange: PostgresArrayEncodable where Bound: PostgresRangeBound {
+    public static var psqlArrayType: PostgresDataType { Bound.rangeType.arrayType! }
+}
```

Comment:
Having the separation into `PostgresRangeEncodable` and `PostgresRangeDecodable` will make this much nicer.